wpengine/faustjs
# Core Quote Block

## Tasks


## Description

This PR implements the `CoreQuote` block in the blocks library.

Headless site on the **left**, WordPress on the **right**:
<img width="2560" alt="Screenshot 2023-06-07 at 9 16 09 AM" src="https://github.com/wpengine/faustjs/assets/5946219/4baf18bb-55a5-41b8-b0ab-68ed0c403b86">



## Related Issue(s):



## Testing

[You can reference this commit for the testing changes.](https://github.com/wpengine/faustjs/commit/4f254242dd3a4207c3c0feb1d9603df24a43ee60)


1. Follow instructions to configure the `blocks` package in the example project. https://faustjs.org/docs/gutenberg/getting-started
2. Generate the global stylessheet and import it into the project. `faust generateGlobalStylesheet`.
3. Import your `theme.json` from your WordPress template (I'm using `twentytwentytwo`)
    ```_app.js
    import { WordPressBlocksProvider, fromThemeJson } from '@faustwp/blocks';
    import themeJson from '../theme.json';
    import '../globalStylesheet.css';
    
    export default function MyApp({ Component, pageProps }) {
      const router = useRouter();
      return (
        <FaustProvider pageProps={pageProps}>
          <WordPressBlocksProvider
            config={{
              blocks,
              theme: fromThemeJson(themeJson),
            }}>
            <Component {...pageProps} key={router.asPath} />
          </WordPressBlocksProvider>
        </FaustProvider>
      );
    }
    ```
4. Import the blocks:
    ```js
    import { coreBlocks } from '@faustwp/blocks';
    
    export default {
      ...coreBlocks,
    };
    ```
5. Make sure you use the block fragment in the page query:
    ```graphql
    ${components.CoreQuote.fragments.entry}
    ...
    editorBlocks {
            __typename
            renderedHtml
            id: clientId
            parentClientId
            ...${components.CoreQuote.fragments.key}
          }
    ```
6. Use the `WordPressBlocksViewer`:
    ```jsx
    <Container>
      <ContentWrapper>
        <WordPressBlocksViewer blocks={blocks}/>
      </ContentWrapper>
    </Container>
    ```
7. Go to WordPress site and create several quote blocks combining styles and colors. 



## Screenshots



## Documentation Changes



## Dependant PRs
